sdk/trace/trace_test.go

```diff
@@ -2177,6 +2182,124 @@ func TestAddLinkToNonRecordingSpan(t *testing.T) {
 	}
 }
 
+func TestSelfObservabilty(t *testing.T) {
+	var got metricdata.ResourceMetrics
+
+	t.Setenv("OTEL_GO_X_SELF_OBSERVABILITY", "True")
+	prev := otel.GetMeterProvider()
+	defer otel.SetMeterProvider(prev)
+	r := metric.NewManualReader()
+	mp := metric.NewMeterProvider(metric.WithReader(r))
+	otel.SetMeterProvider(mp)
```

Comment:
Maybe a dumb question, but will setting the global cause issues when running tests in parallel?